packages/forms/src/widgets/DatalistWidget/DatalistWidget.js

```diff
@@ -151,12 +151,16 @@ class DatalistWidget extends React.Component {
 		};
 	}
 
-	onBlur() {
+	onBlur(event) {
 		if (this.props.options &&
-			this.props.options.restricted && this.state.initalItems.indexOf(this.state.value) === -1) {
+			this.props.options.restricted &&
```

Comment:
Accessing nested properties has a performance cost. I think it could be better like that : 

```
const { options } = this.props;
[...]
if(options && options.restricted
[...]
```